tasks/Task.cpp

```diff
@@ -13,37 +47,650 @@ Task::~Task()
 {
 }
 
-
-
 /// The following lines are template definitions for the various state machine
 // hooks defined by Orocos::RTT. See Task.hpp for more detailed
 // documentation about them.
 
 bool Task::configureHook()
 {
-    if (! TaskBase::configureHook())
+    if (!TaskBase::configureHook()) {
         return false;
+    }
+
+    try {
+        m_system = Arena::OpenSystem();
+        connectToCamera();
+        switchOverAccess();
+        configureCamera();
+    }
+    catch (GenICam::GenericException& ge) {
+        LOG_ERROR_S << "GenICam exception thrown: " << ge.what() << endl;
```

Comment:
*Now* I see why you had to close the system ... because you're not cleaning up after yourself in case of an exception.

I would recommend being *very* anal about the whole thing and create two RAII containers, one for the system, and one for the device. Change connectToCamera to return the device (it's better that way anyways), and store them in the RAII containers in the block.

Now, only just before the return true, do you move their ownership to the class with release()